lib/cdo/crowdin/utils.rb

```diff
@@ -0,0 +1,86 @@
+require 'json'
+require 'parallel'
+
+module Crowdin
+  class Utils
+    # @param project [Crowdin::Project]
+    # @param options [Hash, nil]
+    # @param options.changes_json [String, nil] path to file where files with
+    #  changes will be written out in JSON format
+    # @param options.etags_json [String, nil] path to file where etags will be
+    #  written out in JSON format
+    # @param options.locales_dir [String, nil] path to directory where changed
+    #  files should be downloaded
+    # @param options.logger [Logger, nil]
+    def initialize(project, options={})
+      @project = project
+      @changes_json = options.fetch(:changes_json, "/tmp/#{project.id}_changes.json")
+      @etags_json = options.fetch(:etags_json, "/tmp/#{project.id}_etags.json")
+      @locales_dir = options.fetch(:locales_dir, "/tmp/locales")
+      @logger = options.fetch(:logger, Logger.new(STDOUT))
+    end
+
+    # Fetch from Crowdin a list of files changed since the last sync. Uses
+    # etags sourced from the @etags_json file to define what we mean by "since
+    # the last sync," and writes the results out to @changes_json.
+    def fetch_changes
+      etags = File.exist?(@etags_json) ? JSON.parse(File.read(@etags_json)) : {}
+      changes = {}
+
+      languages = @project.languages
+      num_languages = languages.length
+      languages.each_with_index do |language, i|
+        code = language["code"]
```

Comment:
As someone who hasn't been in this domain long, changing `code` to `language_code` would make the rest of this method much easier to read.  At first I bounced off of lines like this:

```
etag = etags[code].fetch(file, nil)
response = @project.export_file(file, code, etag)
```

And there's a tiny risk of confusing language code with response code, since we're concerned with both here.